tools/tasks/seed/check.tools.ts

```diff
@@ -0,0 +1,30 @@
+import * as util from 'gulp-util';
+import * as runSequence from 'run-sequence';
+
+import { join } from 'path';
+
+let fs = require('fs');
+
+
+// if gulpfile.ts has been compiled then we need to rebuild the toolchain
+
+export = (done: any) => {
+
+  let checkFile = join(process.cwd(), 'tools', 'config.js');
+
+  // need to require the build.toolchain task as it won't be able to run after we run clear.files
+  let buildTools = require('./build.tools');
+
+  try {
+    fs.accessSync(checkFile, fs.F_OK);
+    util.log('Gulpfile has previously been compiled, rebuilding toolchain');
+    runSequence('clean.tools');
```

Comment:
ah right, those shenanigans are because once you run clean.tools you can't run any tasks again until the tools have been rebuilt. now that i reverted to the old build.tools there is a similar issue loading the templateLocals